cumulus_library/databases.py

```diff
@@ -41,6 +41,51 @@ def fetchall(self) -> Optional[list[list]]:
         pass
 
 
+class DatabaseParser(abc.ABC):
+    """Parses information_schema results from a database"""
+
+    def _parse_found_schema(self, expected: dict[dict[list]], schema: list[tuple]):
+        """Checks for presence of field for each column in a table
+
+        This naieve method is a first pass, ignoring complexities of differing
+        database variable types, just iterating through looking for column names.
+
+        TODO: on a per database instance, consider a more nuanced approach
+        if needed
+        """
+        output = {}
+        for column, _ in expected.items():
+            output[column] = {}
+            if col_schema := schema[column.lower()]:
+                # is this field nested?
+                if len(expected[column]) > 0:
+                    for field in expected[column]:
+                        col_schema = col_schema.split(field, 1)
+                        if len(col_schema) != 2:
+                            output[column][field] = False
+                            col_schema = col_schema[0]
+                        else:
+                            output[column][field] = True
+                            col_schema = col_schema[1]
+                # otherwise this field is a bare type
+                else:
+                    output[column] = True
+            else:
+                for field in expected[column]:
+                    output[column][field] = False
+        return output
```

Comment:
I assume this was tested on Athena at some point yeah?

nit: It's not the prettiest little function :smile: Maybe some example input in the docstring or comments would help it be more readable - as you look at it, you can scan above to see, "ah yeah, this would parse that bit" or something.